src/file-selector.ts

```diff
@@ -121,7 +121,7 @@ function flatten<T>(items: any[]): T[] {
 }
 
 function fromDataTransferItem(item: DataTransferItem, entry?: FileSystemEntry | null) {
-    if (typeof (item as any).getAsFileSystemHandle === 'function') {
+    if (typeof window !== 'undefined' && window.isSecureContext && typeof (item as any).getAsFileSystemHandle === 'function') {
```

Comment:
Let's add comment with a link to the issue that is responsible for this in case anyone ever touches this code and wonders why we need such an elaborate guard clause.
```suggestion
    // Check if we're on a secure context due to a bug in Chrome which occasionally causes crashes when calling this API.
    // See:
    // - https://issues.chromium.org/issues/40186242
    // - https://github.com/react-dropzone/react-dropzone/issues/1397
    if (typeof window !== 'undefined' && window.isSecureContext && typeof (item as any).getAsFileSystemHandle === 'function') {
```

Also do we need to guard against `window` here? I don't see this code ever being called in a server context. Perhaps we can use [`globalThis`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis)? It's the sort of modern cross-compatible `window` these days:

> The `globalThis` property provides a standard way of accessing the global `this` value (and hence the global object itself) across environments. Unlike similar properties such as `window` and `self`, it's guaranteed to work in window and non-window contexts. In this way, you can access the global object in a consistent manner without having to know which environment the code is being run in. To help you remember the name, just remember that in global scope the `this` value is `globalThis`.


```suggestion
    if (globalThis.isSecureContext && typeof (item as any).getAsFileSystemHandle === 'function') {
```

Reply:
> Let's add comment with a link to the issue that is responsible for this in case anyone ever touches this code and wonders why we need such an elaborate guard clause.

Will do.

> Also do we need to guard against window here? I don't see this code ever being called in a server context. Perhaps we can use [globalThis](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis)? It's the sort of modern cross-compatible window these days:

Checkout https://github.com/react-dropzone/react-dropzone/commit/c17831433243e28e6ada34cccbc14865877be1e0.

Reply:
Looking at that commit I still think we can get rid of this guard clause and reduce this to `globalThis.isSecureContext`.

The linked code is executed in an SSR environment since it's part of a React hook, but the code in this PR never will be because before calling `fromDataTransferItem()` so you'd actually need to be in a browser selecting a file, which will never happen in SSR environments such as Node.js, so we can safely omit these checks.